lib/capistrano/configuration.rb

```diff
@@ -47,10 +47,14 @@ def remove(key, *values)
     def any?(key)
       value = fetch(key)
       if value && value.respond_to?(:any?)
```

Comment:
Would `value.method(:any?).parameters == []` be a better way of handling this instead of the rescue? Lighter-weight and only saves us from the specific case we care about.

Also, is there any way to write a reliable test for both cases?

Reply:
> `value.method(:any?).parameters == []`

That doesn't work for `Array#any?` in Ruby 2.5.1:

```
>> [].method(:any?).parameters
[
    [0] [
        [0] :rest
    ]
]
```

And it gives a different result for forwarded methods:

```
>> class Foo
>>   extend Forwardable
>>   def_delegators :@array, :any?
>> 
>>   def initialize
>>     @array = []
>>   end
>> end
:initialize
>> Foo.new.method(:any?).parameters
[
    [0] [
        [0] :rest,
        [1] :args
    ],
    [1] [
        [0] :block,
        [1] :block
    ]
]
```

So basically we can't trust `arity` or `parameters`. 😄

Reply:
> Also, is there any way to write a reliable test for both cases?

Yes, I can rewrite the test so that it does not use `stub` (which makes it inconsistent based on the mocha version).